messages/src/lib.rs

```diff
@@ -426,16 +431,87 @@ impl ModuleId {
         }
     }
 
+    /// Convenience method to build a `ModuleId` that selects no transceivers on
+    /// the given FPGA.
+    pub const fn empty(fpga_id: u8) -> Self {
+        Self {
+            fpga_id,
+            ports: PortMask::empty(),
+        }
+    }
+
     /// Return `true` if the provided index is contained in set of addressed
     /// modules.
     pub const fn contains(&self, ix: u8) -> bool {
         self.ports.contains(ix)
     }
 }
 
+/// The MAC address allocation for the target system.
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize, SerializedSize)]
+pub struct MacAddrs {
+    /// The first valid MAC address for the system.
+    pub base_mac: [u8; 6],
+    /// The number of available addresses following `base_mac`.
+    pub count: u16,
+    /// The stride pattern used to compute the next available MAC address.
+    ///
+    /// This is used in situations in which subsystems may make assumptions
+    /// about spacing between addresses. At the time of writing, this applies to
+    /// the Chelsio T6 NICs.
+    pub stride: u8,
+}
+
+impl MacAddrs {
+    /// Return an iterator over the MAC addresses in `self`.
+    pub const fn iter(&self) -> MacAddrIter {
+        MacAddrIter {
+            mac_addrs: *self,
+            current: 0,
+        }
+    }
+
+    /// Return the `n`th address in the range of MACs provided by `self`.
+    ///
+    /// If `n` is outside of the range of `self`, `None` is returned.
+    pub fn nth(&self, n: u16) -> Option<[u8; 6]> {
+        if n >= self.count {
+            return None;
+        }
+        let base_n = u16::from_be_bytes([self.base_mac[4], self.base_mac[5]]);
+        let shift = (base_n + n * u16::from(self.stride)).to_be_bytes();
```

Comment:
How about a `checked_add` here to catch avoid potential overflow?

Reply:
Nice, will do

Reply:
I guess a `checked_mul` would be helpful too :)

Reply:
Good thing I looked again. This implementation is not correct. I believe it basically assumes that the base address has zero for the last two octets.

Reply:
I guess this does raise the question whether we should check that the resulting MAC always has the Oxide OUI. It seems like should _not_ allow MACs without that, but I'm not sure what to do in that case.

It seems odd, but we could get a base MAC like `a8:40:25:ff:ff:00`. A count of say 257 or more would then result in MACs like `a8:40:26:....`, which seems like something we should avoid.

Reply:
While checking that you don't exceed the OUI is useful, I don't think it's appropriate to hard code the list of Oxide OUIs. We had similar discussions around this with @citrus-it and others when we're looking at this in the context of Hubris and the host OS. The valid set of OUIs we have will change over time and it seems like we shouldn't rely on a hardware update for that. The block overlap is an important part and something we need to track in manufacturing assignment (cc @jclulow).

Reply:
Sounds good. Current plan is to keep things within the _existing_ OUI in `base_mac` when computing the `nth` address or in the iterator, but not checking anything about the specific OUI.

Reply:
FWIW in manufacturing we are always assigning 256 addresses for sidecars; each base address is a round number of the form `OO:UU:II:aa:bb:00`.

Reply:
Talked more with Robert here. It seems like a better plan is to fail pretty hard, pretty fast. We'll do some checks in the constructor that the full range is realizable, and if we don't pass some basic sanity checks then we fail right away.